diff --git a/src/afl-fuzz-bitmap.c b/src/afl-fuzz-bitmap.c
index 19cdf511149f..dd7cb510b62c 100644
--- a/src/afl-fuzz-bitmap.c
+++ b/src/afl-fuzz-bitmap.c
@@ -639,7 +639,6 @@ u8 __attribute__((hot)) save_if_interesting(afl_state_t *afl, void *mem,
       if (san_fault == FSRV_RUN_OK) {
 
         if (unlikely(afl->crash_mode)) { ++afl->total_crashes; }
-        return 0;
 
       } else {
 
@@ -934,8 +933,6 @@ may_save_fault:
 
       ++afl->total_crashes;
 
-      if (afl->saved_crashes >= KEEP_UNIQUE_CRASH) { return keeping; }
-
       if (likely(!afl->non_instrumented_mode)) {
 
         if (unlikely(!classified)) {
@@ -947,8 +944,6 @@ may_save_fault:
 
         simplify_trace(afl, afl->fsrv.trace_bits);
 
-        if (!has_new_bits(afl, afl->virgin_crash)) { return keeping; }
-
       }
 
       if (unlikely(!afl->saved_crashes) &&
