diff --git a/fuzzolic/analyze_type.py b/fuzzolic/analyze_type.py
index ea3b4fa24768..c34d74c86e16 100644
--- a/fuzzolic/analyze_type.py
+++ b/fuzzolic/analyze_type.py
@@ -11,7 +11,7 @@ from dataclasses import dataclass
 from collections import defaultdict
 from sortedcontainers import SortedDict, SortedList
 
-import logger
+from . import logger
 
 # OSPREY-style probabilistic weights
 P_UP = 0.8
diff --git a/fuzzolic/binradar.py b/fuzzolic/binradar.py
index 93593a664ff0..f6cbba3f76c7 100644
--- a/fuzzolic/binradar.py
+++ b/fuzzolic/binradar.py
@@ -22,12 +22,12 @@ import fcntl
 from types import TracebackType
 from typing import Dict, List, Tuple, Set, Optional, TextIO, BinaryIO
 
-import analyze_type
-import binradar_verifier
-import binradar_fuzzer
-import binradar_minimizer
-import binradar_utils
-import logger
+from . import analyze_type
+from . import binradar_verifier
+from . import binradar_fuzzer
+from . import binradar_minimizer
+from . import binradar_utils
+from . import logger
 import sbsv
 
 SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
diff --git a/fuzzolic/binradar_fuzzer.py b/fuzzolic/binradar_fuzzer.py
index 5592d83e6035..b5d73769a96b 100644
--- a/fuzzolic/binradar_fuzzer.py
+++ b/fuzzolic/binradar_fuzzer.py
@@ -8,9 +8,8 @@ from typing import List, Set, Tuple, Dict, Optional, Any, TextIO
 
 import sbsv
 
-import logger
-
-import binradar_utils
+from . import binradar_utils
+from . import logger
 
 ROOT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 QEMU_TARGETED_SIMPLE_RELEASE = os.path.join(ROOT_DIR, "LibAFL", "fuzzers", "binary_only", "qemu_targeted_simple", "target", "release", "qemu_targeted_simple")
diff --git a/fuzzolic/binradar_minimizer.py b/fuzzolic/binradar_minimizer.py
index 931c06c539d9..49d16bc4189e 100644
--- a/fuzzolic/binradar_minimizer.py
+++ b/fuzzolic/binradar_minimizer.py
@@ -12,7 +12,7 @@ import time
 
 from typing import List, Dict, Set, Tuple, Any, Optional
 
-import binradar_verifier
+from . import binradar_verifier
 
 ROOT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 QEMU_STACKTRACE_RELEASE = os.path.join(ROOT_DIR, "LibAFL", "fuzzers", "binary_only", "qemu_stacktrace", "target", "release", "qemu_stacktrace")
diff --git a/fuzzolic/binradar_utils.py b/fuzzolic/binradar_utils.py
index 42f8aa400d0f..58fc56173af9 100644
--- a/fuzzolic/binradar_utils.py
+++ b/fuzzolic/binradar_utils.py
@@ -4,7 +4,7 @@ import signal
 import threading
 from typing import List, Set, Tuple, Dict, Optional, Any
 
-import logger
+from . import logger
 
 class ExecutionResult:
     def __init__(self, success: bool, exit_code: int, stdout: str, stderr: str):
diff --git a/fuzzolic/binradar_verifier.py b/fuzzolic/binradar_verifier.py
index be26f4b40173..1724a06b311a 100644
--- a/fuzzolic/binradar_verifier.py
+++ b/fuzzolic/binradar_verifier.py
@@ -8,9 +8,8 @@ from typing import List, Set, Tuple, Dict, Optional, Any, TextIO
 
 import sbsv
 
-import logger
-
-import binradar_utils
+from . import binradar_utils
+from . import logger
 
 ROOT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 QEMU_STACKTRACE_RELEASE = os.path.join(ROOT_DIR, "LibAFL", "fuzzers", "binary_only", "qemu_stacktrace", "target", "release", "qemu_stacktrace")
diff --git a/fuzzolic/executor.py b/fuzzolic/executor.py
index 534f56d64eb4..75f4d2814123 100644
--- a/fuzzolic/executor.py
+++ b/fuzzolic/executor.py
@@ -1,5 +1,6 @@
 #!/usr/bin/python3
 
+import io
 import os
 import sys
 import sbsv
@@ -21,12 +22,11 @@ import struct
 import select
 import threading
 
-import io
-import analyze_type
-import minimizer_qsym
-import minimizer
-import binradar_minimizer
-import logger
+from . import analyze_type
+from . import minimizer_qsym
+from . import minimizer
+from . import binradar_minimizer
+from . import logger
 
 SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
 SOLVER_SMT_BIN = SCRIPT_DIR + '/../solver/build/solver-smt'
diff --git a/fuzzolic/fuzzolic.py b/fuzzolic/fuzzolic.py
index 46c60febdbec..8a92038e953e 100755
--- a/fuzzolic/fuzzolic.py
+++ b/fuzzolic/fuzzolic.py
@@ -2,11 +2,12 @@
 
 import os
 import sys
-import executor
 import signal
 import argparse
 import shutil
 
+from . import executor
+
 ABORTING_COUNT = 0
 
 
diff --git a/fuzzolic/minimizer_qsym.py b/fuzzolic/minimizer_qsym.py
index 9d01aa661d96..66aa555af315 100644
--- a/fuzzolic/minimizer_qsym.py
+++ b/fuzzolic/minimizer_qsym.py
@@ -13,7 +13,7 @@ import struct
 import hashlib
 import os
 import glob
-import logger
+from . import logger
 
 SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
 
diff --git a/pyproject.toml b/pyproject.toml
index fb546dcb453e..f04387bf7f53 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,7 @@
+[build-system]
+requires = ['flit_core >=3.2']
+build-backend = 'flit_core.buildapi'
+
 [project]
 name = "fuzzolic"
 version = "0.1.0"
@@ -5,7 +9,14 @@ description = "Add your description here"
 readme = "README.md"
 requires-python = ">=3.10"
 dependencies = [
-    "pytest>=8.3.5",
     "sbsv>=0.2.3,<0.3.0",
     "sortedcontainers>=2.4.0",
 ]
+
+[project.optional-dependencies]
+test = [
+    "pytest>=8.3.5",
+]
+
+[project.scripts]
+binradar = 'fuzzolic.binradar:main'
