diff --git a/fuzzolic/binradar.py b/fuzzolic/binradar.py
index 93593a664ff0..f6cbba3f76c7 100644
--- a/fuzzolic/binradar.py
+++ b/fuzzolic/binradar.py
@@ -22,11 +22,11 @@ import fcntl
 from types import TracebackType
 from typing import Dict, List, Tuple, Set, Optional, TextIO, BinaryIO
 
-import binradar_verifier
-import binradar_fuzzer
-import binradar_minimizer
-import binradar_utils
-import logger
+from . import binradar_verifier
+from . import binradar_fuzzer
+from . import binradar_minimizer
+from . import binradar_utils
+from . import logger
 import sbsv
 
 SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
diff --git a/fuzzolic/binradar_fuzzer.py b/fuzzolic/binradar_fuzzer.py
index 5592d83e6035..b5d73769a96b 100644
--- a/fuzzolic/binradar_fuzzer.py
+++ b/fuzzolic/binradar_fuzzer.py
@@ -8,9 +8,8 @@ from typing import List, Set, Tuple, Dict, Optional, Any, TextIO
 
 import sbsv
 
-import logger
-
-import binradar_utils
+from . import binradar_utils
+from . import logger
 
 ROOT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 QEMU_TARGETED_SIMPLE_RELEASE = os.path.join(ROOT_DIR, "LibAFL", "fuzzers", "binary_only", "qemu_targeted_simple", "target", "release", "qemu_targeted_simple")
diff --git a/fuzzolic/binradar_minimizer.py b/fuzzolic/binradar_minimizer.py
index 931c06c539d9..49d16bc4189e 100644
--- a/fuzzolic/binradar_minimizer.py
+++ b/fuzzolic/binradar_minimizer.py
@@ -12,7 +12,7 @@ import time
 
 from typing import List, Dict, Set, Tuple, Any, Optional
 
-import binradar_verifier
+from . import binradar_verifier
 
 ROOT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 QEMU_STACKTRACE_RELEASE = os.path.join(ROOT_DIR, "LibAFL", "fuzzers", "binary_only", "qemu_stacktrace", "target", "release", "qemu_stacktrace")
diff --git a/fuzzolic/binradar_utils.py b/fuzzolic/binradar_utils.py
index 42f8aa400d0f..58fc56173af9 100644
--- a/fuzzolic/binradar_utils.py
+++ b/fuzzolic/binradar_utils.py
@@ -4,7 +4,7 @@ import signal
 import threading
 from typing import List, Set, Tuple, Dict, Optional, Any
 
-import logger
+from . import logger
 
 class ExecutionResult:
     def __init__(self, success: bool, exit_code: int, stdout: str, stderr: str):
diff --git a/fuzzolic/binradar_verifier.py b/fuzzolic/binradar_verifier.py
index be26f4b40173..1724a06b311a 100644
--- a/fuzzolic/binradar_verifier.py
+++ b/fuzzolic/binradar_verifier.py
@@ -11,10 +11,9 @@ from typing import List, Set, Tuple, Dict, Optional, Any, TextIO
 
 import sbsv
 
-import logger
-
-import binradar_utils
-from binradar_taosc_predicates import (
+from . import binradar_utils
+from . import logger
+from .binradar_taosc_predicates import (
     CachedSnapshot,
     ParsedPredicate,
     PredicateFamily,
diff --git a/pyproject.toml b/pyproject.toml
index fb546dcb453e..f04387bf7f53 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,7 @@
+[build-system]
+requires = ['flit_core >=3.2']
+build-backend = 'flit_core.buildapi'
+
 [project]
 name = "fuzzolic"
 version = "0.1.0"
@@ -5,7 +9,14 @@ description = "Add your description here"
 readme = "README.md"
 requires-python = ">=3.10"
 dependencies = [
     "pyelftools>=0.31",
-    "pytest>=8.3.5",
     "sbsv>=0.2.3,<0.3.0",
 ]
+
+[project.optional-dependencies]
+test = [
+    "pytest>=8.3.5",
+]
+
+[project.scripts]
+binradar = 'fuzzolic.binradar:main'
