From d457613c546e1e936440d4a9085f5dfea9a76f0a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nguy=E1=BB=85n=20Gia=20Phong?= <cnx@loang.net>
Date: Mon, 18 Nov 2024 15:59:15 +0900
Subject: [PATCH] Fix same_op_2 test expectation

---
 test/regtest/same_op_2.exp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/regtest/same_op_2.exp b/test/regtest/same_op_2.exp
index f99033a..45790b8 100644
--- a/test/regtest/same_op_2.exp
+++ b/test/regtest/same_op_2.exp
@@ -12,6 +12,7 @@ shl $0x7, %rdi
 sar $0x3, %rdi
 pxor %xmm0, %xmm0
 cvtsi2ss %rax, %xmm0
+sqrtss %xmm0, %xmm1
 xor %esi, %esi
 xor %eax, %eax
 PASSED
