From 4e41d765307fae15c903dc6d1141ec85e9047d87 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nguy=E1=BB=85n=20Gia=20Phong?= <cnx@loang.net>
Date: Wed, 22 Oct 2025 16:37:24 +0900
Subject: [PATCH] Support call trampolines in static-PIE linked executables

---
 src/e9tool/e9frontend.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/e9tool/e9frontend.cpp b/src/e9tool/e9frontend.cpp
index a296a71..080e751 100644
--- a/src/e9tool/e9frontend.cpp
+++ b/src/e9tool/e9frontend.cpp
@@ -1022,7 +1022,7 @@ ELF *e9tool::parseELF(const char *filename, intptr_t base)
         end = std::max(end, phdr_end);
         switch (phdr->p_type)
         {
-            case PT_INTERP:
+            case PT_DYNAMIC:
                 if (!exe && !isLibraryFilename(filename))
                     exe = true;
                 dynlink = true;
