From f06525aaf9790b0eecef317e4aaf444189e6042b Mon Sep 17 00:00:00 2001
From: Andrew Haberlandt <ahaberla@andrew.cmu.edu>
Date: Sun, 19 May 2024 06:46:05 +0000
Subject: [PATCH] fix: executor fails to kill solver if it hangs

---
 fuzzolic/executor.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fuzzolic/executor.py b/fuzzolic/executor.py
index 39c6aa6..e660314 100644
--- a/fuzzolic/executor.py
+++ b/fuzzolic/executor.py
@@ -449,7 +449,7 @@ def fuzz_one(self, testcase, target, force_smt=False):
                 print('[FUZZOLIC] Solver is taking too long. Let us stop it.')
                 p_solver.send_signal(signal.SIGUSR2)
                 try:
-                    p_solver.wait(SOLVER_TIMEOUT)
+                    p_solver.wait(SOLVER_TIMEOUT / 1000)
                 except subprocess.TimeoutExpired:
                     print('[FUZZOLIC] Solver will be killed.')
                     p_solver.send_signal(signal.SIGKILL)
