commit bfd24f9cb93fab5317d16c879857221cc70e54c8
Author: Nguyễn Gia Phong <cnx@loang.net>
Date:   2025-04-24 11:25:20 +0900

    Include stdio.h

diff --git a/lib/testcase-list.c b/lib/testcase-list.c
index 0aa223b7a7e4..6457fd5cab87 100644
--- a/lib/testcase-list.c
+++ b/lib/testcase-list.c
@@ -1,4 +1,5 @@
 #include <fts.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include "testcase-list.h"
 
diff --git a/lib/z3-fuzzy-debug-utils.h b/lib/z3-fuzzy-debug-utils.h
index 0f6f3b891a52..7b88add6c294 100644
--- a/lib/z3-fuzzy-debug-utils.h
+++ b/lib/z3-fuzzy-debug-utils.h
@@ -1,3 +1,5 @@
+#include <stdio.h>
+
 static inline void print_index_queue(ast_info_ptr data)
 {
     index_group_t* group;
diff --git a/lib/z3-fuzzy.c b/lib/z3-fuzzy.c
index cac1c329088d..69ecb006fd1f 100644
--- a/lib/z3-fuzzy.c
+++ b/lib/z3-fuzzy.c
@@ -1,6 +1,7 @@
 #define FUZZY_SOURCE
 
 #include <fcntl.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include "gradient_descend.h"
diff --git a/tools/fuzzy-solver-notify.c b/tools/fuzzy-solver-notify.c
index 821d3aaff15f..97abd8f71488 100644
--- a/tools/fuzzy-solver-notify.c
+++ b/tools/fuzzy-solver-notify.c
@@ -1,5 +1,6 @@
 #define FUZZY_SOURCE
 
+#include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
 #include <sys/time.h>
diff --git a/tools/fuzzy-solver-vs-z3.c b/tools/fuzzy-solver-vs-z3.c
index 8803ff390fa2..ba4e8f5b979a 100644
--- a/tools/fuzzy-solver-vs-z3.c
+++ b/tools/fuzzy-solver-vs-z3.c
@@ -1,3 +1,4 @@
+#include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
 #include <sys/time.h>
diff --git a/tools/stats-collection-fuzzy.c b/tools/stats-collection-fuzzy.c
index fdef41761726..c793b1df717d 100644
--- a/tools/stats-collection-fuzzy.c
+++ b/tools/stats-collection-fuzzy.c
@@ -1,3 +1,4 @@
+#include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
 #include <sys/time.h>
diff --git a/tools/stats-collection-z3.c b/tools/stats-collection-z3.c
index 504648a1a6c3..f101eeedb32f 100644
--- a/tools/stats-collection-z3.c
+++ b/tools/stats-collection-z3.c
@@ -1,3 +1,4 @@
+#include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
 #include <sys/time.h>
@@ -105,4 +106,4 @@ int main(int argc, char* argv[])
     Z3_del_context(ctx);
     fclose(log_file);
     return 0;
-}
\ No newline at end of file
+}
