diff --git a/src/afl-showmap.c b/src/afl-showmap.c
index 881ca2a63ffe..a3485b881b3e 100644
--- a/src/afl-showmap.c
+++ b/src/afl-showmap.c
@@ -431,8 +431,8 @@ void pre_afl_fsrv_write_to_testcase(afl_forkserver_t *fsrv, u8 *mem, u32 len) {
 
 /* Execute target application. */
 
-static void showmap_run_target_forkserver(afl_forkserver_t *fsrv, u8 *mem,
-                                          u32 len) {
+static fsrv_run_result_t showmap_run_target_forkserver(afl_forkserver_t *fsrv,
+                                                       u8 *mem, u32 len) {
 
   pre_afl_fsrv_write_to_testcase(fsrv, mem, len);
 
@@ -487,6 +487,7 @@ static void showmap_run_target_forkserver(afl_forkserver_t *fsrv, u8 *mem,
 
   }
 
+  return run_result;
 }
 
 /* Helper to read exactly n bytes from fd */
@@ -1027,7 +1028,11 @@ u32 execute_testcases(u8 *dir) {
 
       }
 
-      showmap_run_target_forkserver(fsrv, in_data, in_len);
+      if (showmap_run_target_forkserver(fsrv, in_data, in_len)
+          == FSRV_RUN_CRASH)
+        snprintf(outfile, sizeof(outfile), "%s/%s.crash", out_file, fn2);
+      else
+        snprintf(outfile, sizeof(outfile), "%s/%s", out_file, fn2);
       ck_free(in_data);
       ++done;
 
@@ -1620,9 +1625,19 @@ int main(int argc, char **argv_orig, char **envp) {
 
     }
 
-    stdin_file = at_file ? strdup(at_file)
-                         : (char *)alloc_printf("%s/.afl-showmap-temp-%u",
-                                                use_dir, (u32)getpid());
+    if (at_file) {
+      stdin_file = strdup(at_file);
+    } else {
+      char* file_ext = get_afl_env("FILE_EXT");
+      if (file_ext)
+        stdin_file =
+            (char *)alloc_printf("%s/.afl-showmap-temp-%u.%s",
+                                 use_dir, (u32)getpid(), file_ext);
+      else
+        stdin_file =
+            (char *)alloc_printf("%s/.afl-showmap-temp-%u",
+                                 use_dir, (u32)getpid());
+    }
     unlink(stdin_file);
 
     // If @@ are in the target args, replace them and also set use_stdin=false.
