commit 861d355ae75fc25108650b87a99caac9bdc31b70
Author: Nguyễn Gia Phong <cnx@loang.net>
Date:   2025-05-08 11:00:18 +0900

    Fix test driver's missing includes

diff --git a/tests/atoi.c b/tests/atoi.c
index 06ba5eb9fa5a..0ca45b18a98a 100644
--- a/tests/atoi.c
+++ b/tests/atoi.c
@@ -1,3 +1,5 @@
+#include <stdlib.h>
+
 int atoi_check(const char* s)
 {
     if (atoi(s) == 1234)
diff --git a/tests/driver.c b/tests/driver.c
index 184f9b2e5af3..7772f37b7ae1 100644
--- a/tests/driver.c
+++ b/tests/driver.c
@@ -1,6 +1,8 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <stdint.h>
+#include <string.h>
+#include <unistd.h>
 #include <assert.h>
 
 int simple_if(uint32_t);
diff --git a/tests/model_strcmp.c b/tests/model_strcmp.c
index cd125bb2a0b3..91044a843dbf 100644
--- a/tests/model_strcmp.c
+++ b/tests/model_strcmp.c
@@ -1,5 +1,6 @@
 #include <unistd.h>
 #include <stdio.h>
+#include <string.h>
 
 int model_strcmp(const char* s1)
 {
diff --git a/tests/model_strncmp.c b/tests/model_strncmp.c
index 86d4482afe1a..ae783597268b 100644
--- a/tests/model_strncmp.c
+++ b/tests/model_strncmp.c
@@ -1,5 +1,6 @@
 #include <unistd.h>
 #include <stdio.h>
+#include <string.h>
 
 int model_strncmp(const char* s1)
 {
